src/components/crown/crownConfig/crownConfig.vue

```diff
@@ -30,6 +30,8 @@
       :dropdown-data="dropdownData"
       :dropdown-initially-open="dropdownInitiallyOpen"
       :node="node"
+      :isOpen="taskDropDownWasClicked"
+      @click="() => { this.taskDropDownWasClicked = true; this.boundaryEventDropDownWasClicked = false; }"
```

Comment:
I would prefer if the component emitted an event that was caught here to be processed. As it is I feel that the crown config has too much coupling to the dropdown component by knowing that interactions with it happen through clicks.

Also, how would you feel about renaming the variables? I feel that including `clicked` in the variable name is a little too specific, particularly when the dropdown is initially open without it having been clicked. How about `taskDropDownOpen` and `boundaryEventDropDownOpen`?

Reply:
It is getting a little tricky. There is a test that expects the Boundary Event dropdown to hide after adding a Boundary Event, but it is staying open when reselecting the task.

Reply:
This is an interesting problem. Generally, when two sibling components need to share some state, I pull that state up into some common parent component and pass it down as props (and then the sibling components no longer need to hold that state as internal data at all).

However, we don't want to bloat the crown config component with state that really only belongs to these two dropdowns. How would you feel about creating another component that wraps these two to allow sharing this state?

Reply:
Moving them to a shared component works great, except for the logic where the Task Dropdown is open initially. The crownConfig needs to retain that information.